gensim/models/word2vec.py

```diff
@@ -1610,13 +1612,16 @@ def accuracy(self, questions, restrict_vocab=30000, most_similar=most_similar, u
                     logger.debug("skipping line #%i with OOV words: %s" % (line_no, line.strip()))
                     continue
 
+                original_vocab = self.vocab
+                self.vocab = ok_vocab
                 ignore = set(self.vocab[v].index for v in [a, b, c])  # indexes of words to ignore
                 predicted = None
                 # find the most likely prediction, ignoring OOV words and input words
                 sims = most_similar(self, positive=[b, c], negative=[a], topn=False, restrict_vocab=restrict_vocab)
+                self.vocab = original_vocab
                 for index in matutils.argsort(sims, reverse=True):
```

Comment:
I think there's still a subtle difference with the `most_similar(topn=1)` behavior, especially in the case-flattening scenario: all words that are case-flattened equivalent to the `ignore` words should plausibly be ignored. We don't do this now... but `compute-accuracy.c` does. (Unsure how best to do this. Maybe there needs to be a case-flattened temporary `index2word` list, too.)